java/base-client/src/main/java/no/bankaxept/epayment/client/base/accesstoken/ScheduledAccessTokenPublisher.java

```diff
@@ -29,19 +29,38 @@ public class ScheduledAccessTokenPublisher implements AccessTokenPublisher, Flow
 
     private final Queue<Flow.Subscriber<? super String>> subscribers = new LinkedBlockingQueue<>();
 
+    private String grantType;
+    private String scopes;
+
     public ScheduledAccessTokenPublisher(String uri, String apimKey, String username, String password, Clock clock, ScheduledExecutorService scheduler, HttpClient httpClient) {
         this.uri = uri;
-        this.headers = createHeaders(apimKey, username, password);
+        this.headers = createHeaders(apimKey, username, password, false);
         this.clock = clock;
         this.scheduler = scheduler;
         this.httpClient = httpClient;
         scheduleFetch(0);
     }
 
-    private static LinkedHashMap<String, List<String>> createHeaders(String apimKey, String username, String password) {
+    public ScheduledAccessTokenPublisher(String uri, String id, String secret, String scopes, String grantType, HttpClient httpClient) {
```

Comment:
Maybe it's a bit messy to have two different constructors with different fields like this.. Would it be better to split into different classes or something else?